In [4]:
# check karo libraries installed hai ya nahi
import sys
print("Python version:", sys.version)

# import karke check kar rahe hai ye saare libraries installed hai ya nahi... error to ni de rahe
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

print("Basic libraries - OK ✅")

Python version: 3.13.5 (tags/v3.13.5:6cb20a2, Jun 11 2025, 16:15:46) [MSC v.1943 64 bit (AMD64)]
Basic libraries - OK ✅


In [9]:
# installing yfinance live dataset
!pip install yfinance


[notice] A new release of pip is available: 25.3 -> 26.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [10]:
# importing yfinance dataset
import yfinance as yf

In [11]:
# downloading reliance last 5 year data
df = yf.download("RELIANCE.NS", start="2020-01-01", end="2026-01-01")

[*********************100%***********************]  1 of 1 completed


In [12]:
print("Data ka shape:", df.shape)
print("\nPehli 5 rows:")
df.head()

Data ka shape: (1487, 5)

Pehli 5 rows:


Price,Close,High,Low,Open,Volume
Ticker,RELIANCE.NS,RELIANCE.NS,RELIANCE.NS,RELIANCE.NS,RELIANCE.NS
Date,,,,,
2020-01-01,675.324219,683.152914,673.490123,679.081997,14004468
2020-01-02,686.821228,689.348791,676.397899,676.397899,17710316
2020-01-03,687.648926,689.662017,681.318850,685.792374,20984698
2020-01-06,671.700684,683.510767,670.134933,679.976719,24519177
2020-01-07,682.034546,686.463335,677.068889,679.529321,16683622


In [13]:
import yfinance as yf

# Aaj tak ka live data
df = yf.download("RELIANCE.NS", start="2020-01-01", end="2026-04-05")

print("Data ka shape:", df.shape)
print("\nPehli 5 rows:")
df.head()

[*********************100%***********************]  1 of 1 completed

Data ka shape: (1549, 5)

Pehli 5 rows:


Price,Close,High,Low,Open,Volume
Ticker,RELIANCE.NS,RELIANCE.NS,RELIANCE.NS,RELIANCE.NS,RELIANCE.NS
Date,,,,,
2020-01-01,675.324219,683.152914,673.490123,679.081997,14004468
2020-01-02,686.821167,689.348729,676.397839,676.397839,17710316
2020-01-03,687.648865,689.661956,681.318790,685.792313,20984698
2020-01-06,671.700684,683.510767,670.134933,679.976719,24519177
2020-01-07,682.034607,686.463396,677.068950,679.529381,16683622


In [14]:
# Last 5 rows dekho — latest data confirm karo
print("Last 5 rows:")
df.tail()

Last 5 rows:


Price,Close,High,Low,Open,Volume
Ticker,RELIANCE.NS,RELIANCE.NS,RELIANCE.NS,RELIANCE.NS,RELIANCE.NS
Date,,,,,
2026-03-25,1413.099976,1430.500000,1408.400024,1420.000000,19041331
2026-03-27,1348.099976,1404.699951,1345.000000,1401.000000,23266517
2026-03-30,1343.900024,1365.000000,1334.099976,1335.000000,24393861
2026-04-01,1369.199951,1384.400024,1362.900024,1384.199951,14404619
2026-04-02,1350.500000,1358.199951,1328.000000,1357.000000,21280051


# Day 1 - Data Collection
- Downloaded Reliance Industries stock data using yfinance
- Date range: 2020-01-01 to 2026-04-05
- Total trading days: 1549
- Columns: Open, High, Low, Close, Volume